modules/database/src/main/java/org/eclipse/xpanse/modules/database/service/ServiceDeploymentEntity.java

```diff
@@ -139,9 +141,11 @@ public class ServiceDeploymentEntity extends CreatedModifiedTime {
     @Column(name = "SERVICE_STATE")
     private ServiceState serviceState = ServiceState.NOT_RUNNING;
 
-    /** The id of the Service Template. */
-    @Column(name = "SERVICE_TEMPLATE_ID", nullable = false)
-    private UUID serviceTemplateId;
+    /** The Service Template. */
+    @ManyToOne
+    @JoinColumn(name = "SERVICE_TEMPLATE_ID", nullable = false)
+    @OnDelete(action = OnDeleteAction.CASCADE)
```

Comment:
This can also be a lazy dependency. Pls check other such examples in our codebase